packages/mui-material/src/Typography/typography.spec.tsx

```diff
@@ -55,6 +55,9 @@ const typographyTest = () => {
       <Typography component={CustomComponent} prop1="1" />
       {/* @ts-expect-error */}
       <Typography component={CustomComponent} prop1="1" prop2="12" />
+      <Typography sx={{ typography: { xs: 'body1', sm: 'h2', md: 'h1', lg: 'body2' } }} />
+      {/* @ts-expect-error */}
+      <Typography sx={{ typography: { xs: 'body 1', sm: 'h2', md: 'h1', lg: 'body1' } }} />
```

Comment:
I'm curious if we can eventually use custom typography variants if they are re-declared in `d.ts` files. E.g. will the code above throw an error with this overrides:

```tsx
export interface CustomTypographyVariants {
  ['H1 title']?: React.CSSProperties;
  ['Body L']?: React.CSSProperties;
}

declare module '@mui/material/styles' {
  interface TypographyVariants extends CustomTypographyVariants {}

  interface TypographyVariantsOptions extends CustomTypographyVariants {}
}

declare module '@mui/material/Typography' {
  interface TypographyPropsVariantOverrides {
    ['H1 title']: true;
    ['Body L']: true;
  }
}
```

Or in other words, will TS compiler be "okay" with `body 1` if we declare it beforehand?

Reply:
Hello @goodwin64!  Adding custom typography variants is supported. 

I added module override to unit test to confirm this and also locally tested adding the above code block in `typography.d.ts` 

Pushed up unit test update.

Reply:
Awesome news! Thank you so much!